genson/src/main/java/com/owlike/genson/convert/RuntimeTypeConverter.java

```diff
@@ -58,4 +69,32 @@ private boolean isContainer(T obj) {
                 obj instanceof Collection ||
                 obj instanceof Map;
     }
+
+    private boolean isSimpleType(T obj) {
+      return obj instanceof Boolean ||
+          obj instanceof Number ||
+          obj instanceof String;
+    }
+
+    private void ensureNoCircularRefs(T obj, Context ctx) {
+      if (!isContainer(obj) && !isSimpleType(obj))
```

Comment:
Do you need this check? Seems redundant, as this method is only called if the condition above is true.

Reply:
Yep, it's intentional.  It's just an added check to ensure we're not tracking anything that we don't need to track.  There won't be cycles in Strings, Boolean, Numbers.  Maybe overkill.